File: frontend/src/types/healthCheck.ts
export interface HealthConfig {
  id: string;
  agent_id: string;
  health_url: string;
  interval_sec: number;
  timeout_sec: number;
  enabled: boolean;
  consecutive_failures_threshold: number;
  verified: boolean;
  verified_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface HealthCheckLog {
  id: string;
  agent_id: string;
  checked_at: string;
  status: 'up' | 'down';
  latency_ms: number | null;
  status_code: number | null;
  error_message: string | null;
}

export interface HealthConfigCreateRequest {
  health_url: string;
  interval_sec: number;
  timeout_sec: number;
  enabled: boolean;
  consecutive_failures_threshold?: number;
}

export interface HealthConfigUpdateRequest {
  health_url?: string;
  interval_sec?: number;
  timeout_sec?: number;
  enabled?: boolean;
  consecutive_failures_threshold?: number;
}

File: backend/app/models/health_check.py
import uuid
from datetime import datetime

from sqlalchemy import Boolean, DateTime, ForeignKey, Integer, String, Text, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base


class AgentHealthConfig(Base):
    __tablename__ = "agent_health_configs"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), primary_key=True, default=uuid.uuid4
    )
    agent_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("agents.id", ondelete="CASCADE"),
        nullable=False,
        unique=True,
        index=True,
    )
    health_url: Mapped[str] = mapped_column(String(2048), nullable=False)
    interval_sec: Mapped[int] = mapped_column(Integer, nullable=False, default=60)
    timeout_sec: Mapped[int] = mapped_column(Integer, nullable=False, default=10)
    enabled: Mapped[bool] = mapped_column(Boolean, nullable=False, default=True)
    consecutive_failures_threshold: Mapped[int] = mapped_column(
        Integer, nullable=False, default=3
    )
    verified: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    verified_at: Mapped[datetime | None] = mapped_column(
        DateTime(timezone=True), nullable=True
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )

    agent: Mapped["Agent"] = relationship(  # noqa: F821
        "Agent", back_populates="health_config"
    )


class HealthCheckLog(Base):
    __tablename__ = "health_check_logs"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), primary_key=True, default=uuid.uuid4
    )
    agent_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        ForeignKey("agents.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    checked_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), nullable=False, index=True
    )
    status: Mapped[str] = mapped_column(String(20), nullable=False)  # up / down / error
    latency_ms: Mapped[int | None] = mapped_column(Integer, nullable=True)
    status_code: Mapped[int | None] = mapped_column(Integer, nullable=True)
    error_message: Mapped[str | None] = mapped_column(Text, nullable=True)

    agent: Mapped["Agent"] = relationship(  # noqa: F821
        "Agent", back_populates="health_check_logs"
    )
